orchestrator/src/worker/event_handler/triggers/snos.rs

```diff
@@ -1,176 +1,497 @@
-use std::cmp::min;
-use std::collections::HashSet;
-use std::sync::Arc;
-
-use async_trait::async_trait;
-use color_eyre::eyre::{Result, WrapErr};
-use opentelemetry::KeyValue;
-use starknet::providers::Provider;
-
 use crate::core::config::Config;
 use crate::types::constant::{CAIRO_PIE_FILE_NAME, PROGRAM_OUTPUT_FILE_NAME, SNOS_OUTPUT_FILE_NAME};
 use crate::types::jobs::metadata::{CommonMetadata, JobMetadata, JobSpecificMetadata, SnosMetadata};
 use crate::types::jobs::types::{JobStatus, JobType};
 use crate::utils::metrics::ORCHESTRATOR_METRICS;
 use crate::worker::event_handler::service::JobHandlerService;
 use crate::worker::event_handler::triggers::JobTrigger;
+use async_trait::async_trait;
+use color_eyre::eyre::{Result, WrapErr};
+use opentelemetry::KeyValue;
+use starknet::providers::Provider;
+use std::cmp::{max, min};
+use std::sync::Arc;
 
+/// Triggers the creation of SNOS (Starknet Network Operating System) jobs.
+///
+/// This component is responsible for:
+/// - Determining which blocks need SNOS processing
+/// - Managing job creation within concurrency limits
+/// - Ensuring proper ordering and dependencies between jobs
 pub struct SnosJobTrigger;
 
+/// Represents the boundaries for block processing.
+///
+/// These bounds define the range of blocks that can be processed:
+/// - `lower_limit`: The minimum block number to consider (based on state updates)
+/// - `middle_limit`: The highest block number that has completed SNOS processing (if any)
+/// - `upper_limit`: The maximum block number to process (from sequencer or config limit)
+#[derive(Debug)]
+struct ProcessingBounds {
+    /// Minimum block number to process, derived from completed state updates
+    lower_limit: u64,
+    /// Latest block number that has completed SNOS processing (None if no jobs completed)
+    middle_limit: Option<u64>,
+    /// Maximum block number to process, limited by sequencer or configuration
+    upper_limit: u64,
```

Comment:
> [!NOTE]
> _**Bikeshedding**_ :motorcycle: 

These names feel a bit confusing, maybe we should go for something clearer? Like
```suggestion
    /// Minimum block number to process, derived from completed state updates
    block_n_min: u64,
    /// Latest block number that has completed SNOS processing (None if no jobs completed)
    block_n_completed: Option<u64>,
    /// Maximum block number to process, limited by sequencer or configuration
    block_n_max: u64,
```

Reply:
HAHA, seems fine, have added!